mobile/library/objective-c/EnvoyEngineImpl.mm

```diff
@@ -609,4 +607,16 @@ - (void)terminateNotification:(NSNotification *)notification {
   terminate_engine(_engineHandle, /* release */ false);
 }
 
+- (void)logException:(NSException *)exception {
+  NSLog(@"[Envoy] exception caught: %@", exception);
+
+  NSString *message = [NSString stringWithFormat:@"%@;%@;%@",
+                       exception.name, exception.reason, exception.callStackSymbols.description];
```

Comment:
callStackSymbols may end up being a little bit long but it will be super helpful when localizing the source of a given crash